pkg/controllers/resources/pods/syncer.go

```diff
@@ -274,6 +275,23 @@ func (s *podSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.SyncEv
 			}
 		}
 
+		// propagate pod status changes from host cluster to vcluster when the host pod
+		// is being deleted. We need this because there is a possibility that pod is owned
+		// by a controller which wants the pod status to be either succeeded or failed before
+		// deleting it. But because these status changes are not propogated
+		// to vcluster pod when the host pod is being deleted, vcluster pod's status still
+		// shows as running, hence it cannot be deleted until it has failed or succeeded. This
+		// results in dangling pods on vcluster
+		if !equality.Semantic.DeepEqual(event.Virtual.Status, event.Host.Status) {
+			updated := event.Virtual.DeepCopy()
+			updated.Status = *event.Host.Status.DeepCopy()
+			ctx.Log.Infof("update virtual pod %s, because status has changed", event.Virtual.Name)
+			err := ctx.VirtualClient.Status().Update(ctx, updated)
+			if err != nil && !kerrors.IsNotFound(err) {
+				return ctrl.Result{}, err
+			}
+		}
```

Comment:
I wonder if a straight status update is desired, or the patch + diff logic [down here](https://github.com/loft-sh/vcluster/pull/2427/files#diff-31528d4b6188bc9038b3a499cb5ff53a3b7d184f62fb9ccef03c9e5cfb2c78aeR356-R375) would also fix the issue. In this logic branch that update logic is skipped.

It may be as simple as removing the [return here](https://github.com/loft-sh/vcluster/pull/2427/files#diff-31528d4b6188bc9038b3a499cb5ff53a3b7d184f62fb9ccef03c9e5cfb2c78aeR295), but could have other side effects.

Reply:
@lizardruss we can remove the `return` statement, but it will unnecessarily patch all the other updates as well to the pod spec. Question is do we want to patch other updates to spec/metdata if the host pod is being deleted? AFAIU the check of `if event.Host.DeletionTimestamp != nil` is to identify if we should delete the virtual pod OR process the other updates to virtual pod/host pod.

Reply:
Maybe. One example might be patching to remove finalizers, but seems like we would have come across that case before.

Reply:
I tried this approach of removing finalizers , but that broke other conformance tests, hence discarded that approach.

Reply:
Okay, this is what I did in the latest commit - https://github.com/loft-sh/vcluster/pull/2427/commits/00fd0bd956beb1a9bff43d5de0943d7adab93f29
- Patch just the status in podTranslator.Diff when host.DeletionTimestamp is not nil
- skip all the other updates apart from status when host.DeletionTimeStamp is not nil

Let me know if you think this is good.